### DISCUSSION TITLE
46. OSI 각 계층에서 사용되는 데이터 단위를 간략하게 설명해주세요.

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#47] [General]
.

**Comment by eun-byeol** [comment]
물리 계층은 `bit`, 데이터 링크 계층은 `frame`, 네트워크 계층은 `packet`, 전송 계층은 `segment`(TCP의 경우 segment, UDP의 경우 datagram), 세션/표현/응용 계층은 `data`(message로 하기도 함) 입니다.

---

> PDU(Protocol Data Unit) : 상위 계층이 전달한 데이터에 붙는 제어정보

- `bit` : 0, 1로 표현되는 이진 데이터
- `frame` : 송수신자 주소(MAC 주소), 오류 검출 코드(CRC), 시작과 끝을 나타내는 플래그 포함
- `packet` : 출도착지 주소(IP 주소), 생명주기(TTL), 프로토콜 포함
- `segment` : TCP 프로토콜의 경우 해당. 송수신자 포트 번호, 순서 번호(sequence number), 확인 응답 번호(acknowledgment number), 윈도우 크기(window size) 포함
- `data` : 사용자가 이해할 수 있는 형태의 정보. 세션 ID, 암호화나 압축 정보, 애플리케이션 프로토콜 정보 포함

**Reply by cutehumanS2** [reply]
PDU의 정의를 설명할 때, "'**각 계층에서 송수신되는 메시지의 단위**'로 상위 계층에서 전달받은 데이터를 포함하며, 제어 정보를 추가하거나 필요한 변환을 거쳐 생성됩니다."라고 해야 좀 더 완전해질 것 같아요~~
